core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -85,7 +85,19 @@ private[spark] class CoarseGrainedExecutorBackend(
 
     logInfo("Connecting to driver: " + driverUrl)
     try {
-      if (PlatformDependent.directBufferPreferred() &&
+      // The following logic originally comes from the constructor of
+      // org.apache.spark.network.client.TransportClientFactory
+      val sharedByteBufAllocators =
+        env.conf.getBoolean("spark.network.sharedByteBufAllocators.enabled", true)
+      val preferDirectBufsForSharedByteBufAllocators =
+        env.conf.getBoolean("spark.network.io.preferDirectBufs", true)
+      val preferDirectBufs =
+        env.conf.getBoolean("spark.shuffle.io.preferDirectBufs", true)
```

Comment:
Did you see where is this conf accessed? I didn't find it used anywhere.

Reply:
it's a combination of ("spark", module, "io.preferDirectBufs"), for shuffle client, the module is "shuffle", see `org.apache.spark.network.util.TransportConf`